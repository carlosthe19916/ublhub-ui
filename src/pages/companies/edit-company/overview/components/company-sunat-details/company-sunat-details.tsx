import React from "react";
import {
  Card,
  CardTitle,
  CardBody,
  DescriptionList,
  DescriptionListGroup,
  DescriptionListTerm,
  DescriptionListDescription,
} from "@patternfly/react-core";
import { Company } from "api/models";

export interface CompanySunatDetailsProps {
  company: Company;
}

export const CompanySunatDetails: React.FC<CompanySunatDetailsProps> = ({
  company,
}) => {
  return (
    <Card>
      <CardTitle>SUNAT</CardTitle>
      <CardBody>
        <DescriptionList
          columnModifier={{
            default: "2Col",
          }}
        >
          <DescriptionListGroup>
            <DescriptionListTerm>Factura URL</DescriptionListTerm>
            <DescriptionListDescription>
              {company.webServices.factura}
            </DescriptionListDescription>
          </DescriptionListGroup>
          <DescriptionListGroup>
            <DescriptionListTerm>Retención URL</DescriptionListTerm>
            <DescriptionListDescription>
              {company.webServices.retenciones}
            </DescriptionListDescription>
          </DescriptionListGroup>
          <DescriptionListGroup>
            <DescriptionListTerm>Guía remisión URL</DescriptionListTerm>
            <DescriptionListDescription>
              {company.webServices.guia}
            </DescriptionListDescription>
          </DescriptionListGroup>
          <DescriptionListGroup>
            <DescriptionListTerm>Username</DescriptionListTerm>
            <DescriptionListDescription>
              {company.credentials.username}
            </DescriptionListDescription>
          </DescriptionListGroup>
        </DescriptionList>
      </CardBody>
    </Card>
  );
};
