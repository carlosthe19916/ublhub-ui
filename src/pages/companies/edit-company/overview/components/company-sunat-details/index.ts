export { CompanySunatDetails } from "./company-sunat-details";
