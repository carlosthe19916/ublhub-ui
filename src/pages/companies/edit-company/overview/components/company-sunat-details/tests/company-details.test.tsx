import React from "react";
import { shallow } from "enzyme";
import { CompanySunatDetails } from "../company-sunat-details";

describe("CompanySunatDetails", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <CompanySunatDetails
        company={{
          name: "my company",
          description: "my description",
          webServices: {
            factura: "http://url1.com",
            guia: "http://url2.com",
            retenciones: "http://url3.com",
          },
          credentials: {
            username: "myUsername",
            password: "myPassword",
          },
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
