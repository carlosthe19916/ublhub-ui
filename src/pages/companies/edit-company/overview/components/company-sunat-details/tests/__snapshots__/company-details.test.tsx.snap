// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanySunatDetails Renders without crashing 1`] = `
<Card>
  <CardTitle>
    SUNAT
  </CardTitle>
  <CardBody>
    <DescriptionList
      columnModifier={
        Object {
          "default": "2Col",
        }
      }
    >
      <DescriptionListGroup>
        <DescriptionListTerm>
          Factura URL
        </DescriptionListTerm>
        <DescriptionListDescription>
          http://url1.com
        </DescriptionListDescription>
      </DescriptionListGroup>
      <DescriptionListGroup>
        <DescriptionListTerm>
          Retención URL
        </DescriptionListTerm>
        <DescriptionListDescription>
          http://url3.com
        </DescriptionListDescription>
      </DescriptionListGroup>
      <DescriptionListGroup>
        <DescriptionListTerm>
          Guía remisión URL
        </DescriptionListTerm>
        <DescriptionListDescription>
          http://url2.com
        </DescriptionListDescription>
      </DescriptionListGroup>
      <DescriptionListGroup>
        <DescriptionListTerm>
          Username
        </DescriptionListTerm>
        <DescriptionListDescription>
          myUsername
        </DescriptionListDescription>
      </DescriptionListGroup>
    </DescriptionList>
  </CardBody>
</Card>
`;
